### PR TITLE
fixed analyzer issues

### PR DESCRIPTION
**Pull request by viralkachhadiya**
it's fixed to #77425 

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 1]
```diff
@@ -307,14 +307,20 @@ class SlideTransition extends AnimatedWidget {
 /// above:
 ///
 /// ```dart
-/// late AnimationController _controller = AnimationController(
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/transitions.dart:310]: I believe all you need is a final here:

```dart
/// late final AnimationController _controller = AnimationController(
```

File: packages/flutter/lib/src/widgets/transitions.dart [hunk 2]
```diff
@@ -307,14 +307,20 @@ class SlideTransition extends AnimatedWidget {
...
+/// late Animation _animation;
+///
+/// @override
+/// void initState() {
+/// super.initState();
+/// _controller = AnimationController(
+/// duration: const Duration(seconds: 2),
+/// vsync: this,
 /// )..repeat(reverse: true);
-/// late Animation<double> _animation = CurvedAnimation(
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/transitions.dart:314]: same here.

File: packages/flutter/lib/src/widgets/transitions.dart (hunk 1 above)
Comment (viralkachhadiya) [reply packages/flutter/lib/src/widgets/transitions.dart:310]: But still it is give the error on "vsync:this"

**Review by viralkachhadiya — COMMENTED** [review]
Thank you

**Pull request closed without merging**